pgmpy/factors/Factor.py

```diff
@@ -76,7 +76,11 @@ def __init__(self, variables, cardinality, values):
         >>> phi
         <Factor representing phi(x1:2, x2:2, x3:2) at 0x7f8188fcaa90>
         """
+
         values = np.array(values)
+        
+        if values.dtype != int and values.dtype != float:
```

Comment:
@yashu-seth This wouldn't throw any error if we give some other type than `int` or `float`. Both conditions would be false. I think it should be `not(values.dtype != int or values.dtype != float)`